src/altinn/tilganger.ts

```diff
@@ -1,24 +1,55 @@
 import { altinntjeneste, AltinntjenesteId } from './tjenester';
+import { z } from 'zod';
+import useSWR from 'swr';
+import { useEffect, useState } from 'react';
+import * as Sentry from '@sentry/browser';
+import { Simulate } from 'react-dom/test-utils';
+import error = Simulate.error;
 
 const altinnTilgangssøknadUrl = '/min-side-arbeidsgiver/api/altinn-tilgangssoknad';
 
-export interface AltinnTilgangssøknad {
-    orgnr: string;
-    serviceCode: string;
-    serviceEdition: string;
-    status: string;
-    cratedDateTime: string;
-    lastChangedDateTime: string;
-    submitUrl: string;
-}
+const AltinnTilgangssøknad = z.object({
+    orgnr: z.string(),
+    serviceCode: z.string(),
+    serviceEdition: z.string(),
+    status: z.string(),
+    cratedDateTime: z.string(),
+    lastChangedDateTime: z.string(),
+    submitUrl: z.string(),
+});
 
-export const hentAltinnTilgangssøknader = async () => {
-    const response = await fetch(altinnTilgangssøknadUrl);
+export type AltinnTilgangssøknad = z.infer<typeof AltinnTilgangssøknad>;
 
-    if (!response.ok) {
-        throw new Error();
-    }
-    return (await response.json()) as AltinnTilgangssøknad[];
+const AltinnTilgangssøknadResponse = z.array(AltinnTilgangssøknad);
+type AltinnTilgangssøknadResponse = z.infer<typeof AltinnTilgangssøknadResponse>;
+
+export const useAltinnTilgangssøknader = (): AltinnTilgangssøknadResponse => {
+    const [tilgangssøknader, setTilgangssøknader] = useState<AltinnTilgangssøknadResponse>([]);
+    const { data, error } = useSWR(altinnTilgangssøknadUrl, fetcher, {
+        onError: (error) => {
+            Sentry.captureMessage(
+                `hent AltinnTilgangssøknader fra min-side-arbeidsgiver-api feilet med ${
+                    error.status !== undefined ? `${error.status} ${error.statusText}` : error
+                }`
+            );
+        },
+        fallbackData: [],
+        errorRetryInterval: 300,
+    });
+
+    useEffect(() => {
+        setTilgangssøknader(data);
+    }, [JSON.stringify(data), error]);
```

Comment:
Er `error` nødvendig? Kunne brukt `useMemo` i stede for `useEffect` vil jeg tro.

Reply:
Mener altså
```tsx
return useMemo(() => data, [JSON.stringify(data), error])
```
om det skulle være tvil.